todo_app/data/trello_items.py

```diff
@@ -16,16 +15,11 @@ def __init__(self) -> None:
         self.headers = {"Accept": "application/json"}
         self.auth = {"key": self._API_KEY, "token": self._TOKEN}
 
-        # Create a dictionary for the bard lists with list_name: list_id
+        # # Create a dictionary for the bard lists with list_name: list_id
         # This way we can refer to the correct id when moving cards across lists
         # The list_name, list_id values are pretty constant and will remain the same throughout the session.
         # We create this as a variable accessible by all the functions so that we don't have to send the request with each user interaction
-        self.board_lists = self.get_board_lists()
```

Comment:
The test structure for the integration tests kind of forced me to move the the get_board_lists() method call to within the get_cards() function since the test client calls the main app before the requests patches. And if left as it was, the http request to get the list id's gets called within the client and before the stub function.

This change makes the tests pass green, but at the cost of slower performance in the app, since the http request for getting the lists gets made multiple times when get_cards is called.

Reply:
Nice spot - you've picked a good structure so actually this can be a fairly quick fix to get it back to the best of both hopefully - see my comment on test_app